include/zephyr/drivers/gpio.h

```diff
@@ -538,9 +538,14 @@ __subsystem struct gpio_driver_api {
 				   gpio_port_pins_t pins);
 	int (*port_toggle_bits)(const struct device *port,
 				gpio_port_pins_t pins);
+        /* Configure pin interrupt. */
 	int (*pin_interrupt_configure)(const struct device *port,
 				       gpio_pin_t pin,
 				       enum gpio_int_mode, enum gpio_int_trig);
+        /* Enable the pin interrupt. */
+	int (*pin_interrupt_enable)(const struct device *port, gpio_pin_t pin);
```

Comment:
There needs to be a plan to implement this for the other ~2 dozen GPIO drivers otherwise the implementations inconsistent / fragmented.

Also, tests need to be added, etc.

Reply:
Thank you! I put it as a experimental Kconfig, and add tests for it.

Reply:
Please see my [comment](https://github.com/zephyrproject-rtos/zephyr/issues/52186#issuecomment-1332576434) on the RFC.  I think you can avoid expanding the driver side API.

Reply:
Followed comments in the RFC, I think current approach does not changing the driver API. However, I found that maybe I can remove this experimental KConfig, and maybe just add guarding check in the ~2 dozen GPIO drivers, and return `-ENOTSUP` for drivers not supported yet?